tests/readPreference/readpreference-ctor_error-007.phpt

```diff
@@ -0,0 +1,18 @@
+--TEST--
+MongoDB\Driver\ReadPreference construction (combining hedge with primary read preference)
+--FILE--
+<?php
+
+require_once __DIR__ . '/../utils/tools.php';
+
+echo throws(function() {
+    new MongoDB\Driver\ReadPreference(MongoDB\Driver\ReadPreference::RP_PRIMARY, null, ['hedge' => ['enabled' => true]]);
+}, 'MongoDB\Driver\Exception\InvalidArgumentException'), "\n";
+
+?>
+===DONE===
+<?php exit(0); ?>
+--EXPECTF--
+OK: Got MongoDB\Driver\Exception\InvalidArgumentException
+Read preference is not valid
```

Comment:
I think it'd be user-friendly to copy over the exception you have in `php_phongo_readpreference_init_from_hash` so we get a more meaningful error message. This would be consistent with what the constructor was already doing for `maxStalenessSeconds` and `tags`.

I consider the exception from `mongoc_read_prefs_is_valid` to be more of a fail safe in case we missed anything (or someone is using an older PHPC with a newer libmongoc).

While writing this comment, I also realized that `php_phongo_readpreference_init_from_hash` doesn't include a final check for `mongoc_read_prefs_is_valid`. We also don't appear to use the WC valid check at all in WriteConcern.c. I've opened [PHPC-1629](https://jira.mongodb.org/browse/PHPC-1629) to track this.